javascript/ql/src/semmle/javascript/AST.qll

```diff
@@ -181,51 +198,118 @@ class TopLevel extends @toplevel, StmtContainer {
 
 /**
  * A stand-alone file or script originating from an HTML `<script>` element.
+ *
+ * Example:
+ *
+ * ```
+ * <script>
+ * window.done = true;
+ * </script>
+ * ```
  */
 class Script extends TopLevel {
   Script() { this instanceof @script or this instanceof @inline_script }
 }
 
 /**
  * A stand-alone file or an external script originating from an HTML `<script>` element.
+ *
+ * Example:
+ *
+ * ```
+ * #! /usr/bin/node
+ * console.log("Hello, world!");
+ * ```
  */
 class ExternalScript extends @script, Script { }
 
 /**
  * A script embedded inline in an HTML `<script>` element.
+ *
+ * Example:
+ *
+ * ```
+ * <script>
+ * window.done = true;
+ * </script>
+ * ```
  */
 class InlineScript extends @inline_script, Script { }
 
 /**
  * A code snippet originating from an HTML attribute value.
+ *
+ * Examples:
+ *
+ * ```
+ * <div onclick="alert('hi')">Click me</div>
+ * <a href="javascript:alert('hi')">Click me</a>
+ * ```
  */
 class CodeInAttribute extends TopLevel {
   CodeInAttribute() { this instanceof @event_handler or this instanceof @javascript_url }
 }
 
 /**
  * A code snippet originating from an event handler attribute.
+ *
+ * Example:
+ *
+ * ```
+ * <div onclick="alert('hi')">Click me</div>
+ * ```
  */
 class EventHandlerCode extends @event_handler, CodeInAttribute { }
 
 /**
  * A code snippet originating from a URL with the `javascript:` URL scheme.
+ *
+ * Example:
+ *
+ * ```
+ * <a href="javascript:alert('hi')">Click me</a>
+ * ```
  */
 class JavaScriptURL extends @javascript_url, CodeInAttribute { }
 
 /**
  * A toplevel syntactic entity containing Closure-style externs definitions.
+ *
+ * Example:
+ *
+ * <pre>
```

Comment:
Missing backticks?

Reply:
No. We need to use `<pre>` instead of the backticks so we can use the `&#` entity to escape the slash, which otherwise would terminate the comment.